File: 0060-permutation-sequence/0060-permutation-sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> nums(n);
        for (int i = 0; i < n; ++i) nums[i] = i + 1;

        vector<int> factors(n + 1, 1);
        for (int i = 1; i <= n; ++i) factors[i] = factors[i - 1] * i;

        k -= 1; //convert to 0-based index
        string ans;
        for (int len = n; len >= 1; --len) {
            int blockSize = factors[len - 1];
            int index = k / blockSize;
            k = k % blockSize;

            ans.push_back(char('0' + nums[index]));
            nums.erase(nums.begin() + index);
        }
        return ans;
    }
};

File: 0060-permutation-sequence/0060-permutation-sequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        # nums = [i for i in range(n)]
        # for i in range(1,k):
        #     sure
            
            
            
            
        
        
        
        # come lets try, i comment urs first? fk py
        
        
        
        nums = [i+1 for i in range(n)] #get all nums
        factors = [1] * (n+1) #all factors

        for i in range(1,n+1):
            factors[i] = factors[i-1]*i

        k -= 1 #0-based index?
        res = []

        for len in range(n,0,-1):
            block = factors[len-1]
            ind = k // block
            k = k%block
            res.append(str(nums[ind]))
            nums.pop(ind)
            
        return "".join(res)
        
        